scripts/openduck.py

```diff
@@ -0,0 +1,745 @@
+    '''
+    Prepares the system for Amber simulation by parameterizing the ligand and finding the specified interaction
+    between the ligand and protein. The resulting complex is saved to a pickle file, 'complex_system.pickle', and
+    the amber inputs are written for the specified queue system.
+
+    Args:
+        ligand_file (str): Path to the ligand file in .mol2 or .pdb format.
+        protein_file (str): Path to the protein-receptor file in .pdb format.
+        chunk_file (str): Path to the chunk file in .pdb format.
+        interaction (str): String specifying the receptor atom that forms the main interaction with the ligand.
+        HMR (bool): Whether or not to use hydrogen mass repartitioning (HMR) during parameterization.
+        small_molecule_forcefield (str, optional): Force field to use for ligand parameterization. Default is 'SMIRNOFF'.
+        water_ff_str (str, optional): Force field to use for water molecules. Default is 'tip3p.xml'.
+        forcefield_str (str, optional): Force field to use for protein parameterization. Default is 'amber99sb.xml'.
+        ionic_strength (float, optional): Ionic strength of the system in M. Default is 0.1 M.
+        box_buffer_distance (float, optional): Distance in Angstroms to buffer the protein-ligand complex from the edges of the simulation box. Default is 10.
+        waters_to_retain (str, optional): Path to the file containing the water molecules to retain during simulation. Default is "waters_to_retain.pdb".
+        seed (str, optional): Seed value for random number generation during parameterization. Default is '-1'.
+        fix_ligand_file (bool, optional): Whether or not to fix the ligand file during parameterization. Default is False.
+    '''
+    from duck.steps.parametrize import prepare_system
+    from duck.utils.cal_ints import find_interaction
+    from duck.utils.amber_inputs import Amber_templates
+    # Parameterize the ligand
+    prepare_system(ligand_file, chunk_file, forcefield_str=forcefield_str,
+                   hmr=HMR, small_molecule_ff=small_molecule_forcefield, water_ff_str = water_ff_str,
+                   box_buffer_distance = box_buffer_distance, ionicStrength = ionic_strength, waters_to_retain=waters_to_retain, fix_ligand_file=fix_ligand_file)
+    
+    # Now find the interaction and save to a file
+    results = find_interaction(interaction, protein_file)
+    print(results) # what happens to these?
+    with open('complex_system.pickle', 'rb') as f:
+        p = pickle.load(f) + results
+    with open('complex_system.pickle', 'wb') as f:
+        pickle.dump(p, f, protocol=pickle.HIGHEST_PROTOCOL)
+    #p = (parmed_structure, prot_index, ligand_index, pairmeandistance)
```

Comment:
remove

Reply:
It is not really a commented line of code, but more of a tip for whoever is reading the code to understand what is there inside the 'p' variable. I'll rephrase it. I think its worth keeping the explanation around as it is otherwise very obfuscated.